lib/std/compress/zlib.zig

```diff
@@ -102,6 +189,20 @@ fn testReader(data: []const u8, expected: []const u8) !void {
     try testing.expectEqualSlices(u8, expected, buf);
 }
 
+fn testWriter(data: []const u8, expected: []const u8) !void {
```

Comment:
This function is unused. Could you add a small test, so things are ci-tested?

Reply:
I've been having trouble writing tests, it seems out deflate algorithm doesn't match the output of zlib's. It still produces good compression and it's valid so it can be decompressed, but the actual byte stream it produces doesn't match so the tests fail

Reply:
Just doing a round trip test instead of directly testing the output of the compressor. Should work :slightly_smiling_face: